src/server/generateEmail/generateEmail.ts

```diff
@@ -0,0 +1,31 @@
+import ReactDOMServer from 'react-dom/server';
+import fs from 'fs';
+import { deafultHTML } from './defaultHTML';
+
+interface GenerateEmailOptions {
+  baseTemplate?: string;
+  baseStyles?: string;
+}
+
+export const generateEmail = (
+  jsxElement: React.ReactElement,
+  options?: GenerateEmailOptions,
+): string => {
+  let baseHTML = '';
+  options?.baseTemplate
+    ? (baseHTML = fs.readFileSync(options.baseTemplate, 'utf8')).toString()
+    : (baseHTML = deafultHTML);
+
+  let baseCSS = '';
+  options?.baseStyles && (baseCSS = fs.readFileSync(options.baseStyles, 'utf8')).toString();
+
+  const JSXtoHTML = ReactDOMServer.renderToStaticMarkup(jsxElement);
+
+  const styleEndPos = baseHTML.indexOf('</style>');
```

Comment:
Let's not pick the `style` element, uses of the library might pass templates which have their own `style` elements. Introduce a variable like 'EMAIL_BASE_STYLES` and `EMAIL_BODY_CONTENT` ( for HTML part ) and replace there with the styles and `JSXtoHTML`

Reply:
done